database/guests.ts

```diff
@@ -1,44 +1,55 @@
-declare module globalThis {
-  let guestsDatabase: Guest[];
-}
-
-// Initialize database to empty array only once
-//
-// TODO: Add a full PostgreSQL database connection
-if (!('guestsDatabase' in globalThis)) {
-  globalThis.guestsDatabase = [];
-}
-
-const guests = globalThis.guestsDatabase;
-
-console.log('guests in db', guests);
-
-export type Guest = {
-  id: number;
-  firstName: string;
-  lastName: string;
-  attending: boolean;
-};
-
-export function getGuests() {
+import { Guest } from '../migrations/00000-createTableGuests';
+import { sql } from './connect';
+
+export const getGuests = async () => {
+  const guests = await sql<Guest[]>`
+    SELECT
+      *
+    FROM
+      guests
+  `;
   return guests;
-}
-
-export function getGuest(id: number) {
-  return guests.find((guest: Guest) => Number(guest.id) === Number(id));
-}
+};
 
-export function addGuest(guest: Guest) {
-  guests.push(guest);
-  return guests;
-}
+export const getGuest = async (id: number) => {
+  const [guest] = await sql<Guest[]>`
+    SELECT
+      *
+    FROM
+      guests
+    WHERE
+      id = ${id}
+  `;
+  return guest;
+};
 
-export function deleteGuest(id: number) {
-  const index = guests.findIndex((guest) => guest.id === id);
+export const addGuest = async (newGuest: Omit<Guest, 'id'>) => {
+  const [guest] = await sql<Guest[]>`
+    INSERT INTO
+      guests (
+        first_name,
+        last_name,
+        attending
+      )
+    VALUES
+      (
+        ${newGuest.firstName},
+        ${newGuest.lastName},
+        ${newGuest.attending}
+      )
+    RETURNING
+      guests.*
+  `;
+  return guest;
+};
 
-  if (index === -1) {
-    return;
-  }
-  guests.splice(index, 1);
-  return guests;
-}
+export const deleteGuest = async (removeGuest: Pick<Guest, 'id'>) => {
```

Comment:
- the `removeGuest` name can be improved
- however, if we just change to `id: Guest['id']` (a string parameter instead of a partial object), then we avoid thinking about new naming

Reply:
Updated, and now only uses the `id` instead of an `object`. [a2fde60](https://github.com/upleveled/expo-example-spring-2024-atvie/pull/4/commits/a2fde607a42aa392b94c3612b4662b8c5eb572f5)

Reply:
new version:

```ts
export const deleteGuestInsecure = async (id: number) => {
```

I think our conventions call the parameter `guestId` rather than `id`

(check all other database query functions for other naming changes like this)